go/vt/srvtopo/resilient_server.go

```diff
@@ -207,53 +207,51 @@ func (server *ResilientServer) GetSrvKeyspaceNames(ctx context.Context, cell str
 	}
 	server.mutex.Unlock()
 
-	// Lock the entry, and do everything holding the lock.  This
-	// means two concurrent requests will only issue one
-	// underlying query.
-	entry.mutex.Lock()
-	defer entry.mutex.Unlock()
-
 	// If it is not time to check again, then return either the cached
 	// value or the cached error
 	cacheValid := entry.value != nil && time.Since(entry.insertionTime) < server.cacheTTL
```

Comment:
You can't access any value inside entry if you don't have the entry.mutex.Lock()...